ckanfunctionaltests/api/conftest.py

```diff
@@ -29,6 +33,23 @@ def random_org_slug(base_url, rsession):
 
 @pytest.fixture()
 def random_pkg_slug(base_url, rsession):
-    response = rsession.get(f"{base_url}/action/package_list")
+    # make do with one of the first 200 because the full list is big & slow
+    response = rsession.get(f"{base_url}/action/package_list?limit=200")
     assert response.status_code == 200
-    return _random.choice(response.json()["result"])
+
+    suitable_names = tuple(
+        name for name in response.json()["result"] if not _uuid_re.fullmatch(name)
```

Comment:
I wonder if rather than using regex to determine if something is a valid uuid that we can use `uuid.UUID` to try to cast it as a uuid and then catch the exception if it fails? 
As there may be different versions of uuids being being in datasets. 

This shows how we might validate the uuid and get the version - https://stackoverflow.com/questions/45943754/how-to-validate-list-of-uuids-and-return-uuid-version.

Reply:
I considered that, but `UUID` accepts a few different representations of uuid and I wanted to be stricter because not everyone consuming this API is necessarily using python & its `UUID`. They could well be relying on the standard 8-4-4-4-12 representation.

Reply:
I believe that you can also specify which UUID version you want to enforce, if its the case that only uuid 4 is being used within CKAN.

Reply:
It's less about the versions, more about the (from https://docs.python.org/3/library/uuid.html#uuid.UUID):

> When a string of hex digits is given, curly braces, hyphens, and a URN prefix are all optional.

not all consumers of the API are necessarily going to be this lenient.